pureconfig/README.md

```diff
@@ -0,0 +1,244 @@
+  
+        type = "direct" // fanout, topic
+        durable = true // default value
+        autoDelete = false // default value
+      }
+      
+      // These properties are used when none properties are passed to the send method.
+      properties {
+        deliveryMode = 2 // this is default value
+        contentType = "text" // default is not set
+        contentEncoding = "UTF8" // default is not set
+        priority = 1 // default is not set
+      }
+    }
+  }
+}
+```
+Checkout [configuration caseclasses](../core/src/main/scala/com/avast/clients/rabbitmq/configuration.scala) for all possible values along
+with their defaults.
+
+
+### Scala example
+
+The example below uses [Monix](https://monix.io/) `Task` however it works with all `F[_] : Effect` (`Task`, Cats-Effect `IO`, `ZIO`, ..?).
+
+```scala
+import java.util.concurrent.ExecutorService
+
+import cats.effect.Resource
+import com.avast.bytes.Bytes
+import com.avast.clients.rabbitmq._
+import com.avast.clients.rabbitmq.api._
+import com.avast.metrics.scalaapi.Monitor
+import com.typesafe.config.ConfigFactory
+import monix.eval._
+import monix.execution.Scheduler
+
+import com.avast.clients.rabbitmq.pureconfig._ // <-- this is needed in order to be able to use `fromConfig` method!
+
+val config = ConfigFactory.load().getConfig("myConfig")
+
+implicit val sch: Scheduler = ??? // required by Task
+val monitor: Monitor = ???
+
+val blockingExecutor: ExecutorService = ???
+
+// see https://typelevel.org/cats-effect/tutorial/tutorial.html#acquiring-and-releasing-resources
+
+val rabbitMQProducer: Resource[Task, RabbitMQProducer[Task, Bytes]] = {
+    for {
+      connection <- RabbitMQConnection.fromConfig[Task](config, blockingExecutor)
+      /*
+        Here you have created the connection; it's shared for all producers/consumers amongst one RabbitMQ server - they will share a single
+        TCP connection but have separated channels.
+        If you expect very high load, you can use separate connections for each producer/consumer, however it's usually not needed.
+       */
+
+      consumer <- connection.newConsumer[Bytes]("testing", monitor) {
+        case delivery: Delivery.Ok[Bytes] =>
+          Task.now(DeliveryResult.Ack)
+
+        case _: Delivery.MalformedContent =>
+          Task.now(DeliveryResult.Reject)
+      }
+
+      producer <- connection.newProducer[Bytes]("testing", monitor)
+    } yield {
+      producer
+    }
+  }
+```
+
+### Different casing support
+
+The default casing for HOCON keys is `camelCase` but it's also possible to use other ones supported by Pureconfig by using:
+```scala
+import com.avast.clients.rabbitmq.pureconfig.implicits.KebabCase._
+```
+
+See all possible values in `com.avast.clients.rabbitmq.pureconfig.implicits`.
+
+### Turning of strict format checking
+
+It's possible to turn off the check of configuration format. You can do that by importing:
+```scala
+import com.avast.clients.rabbitmq.pureconfig.implicits.nonStrict._
```

Comment:
I think "package names" (even though this can be `object` I know) are usually either all lowercase or `PascalCase`, but I've never seen `camelCase`, what do you think?

Reply:
Never? :stuck_out_tongue: `cats.instances.bigDecimal.catsKernelStdGroupForBigDecimal`  
I kind of like it but don't insist on it.